src/lib/support/SerializableIntegerSet.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "SerializableIntegerSet.h"
+
+#include <core/CHIPEncoding.h>
+
+namespace chip {
+
+const char * SerializableU64SetBase::SerializeBase64(char * buf, uint16_t & buflen)
+{
+    char * out = nullptr;
+    size_t len = sizeof(uint64_t) * mNextAvailable;
+    VerifyOrExit(buflen >= SerializedSize(), buflen = SerializedSize());
+    VerifyOrExit(buf != nullptr, buflen = SerializedSize());
+
+    /**
+     *  The data is serialized in LittleEndian byte order in the set. This will enable
+     *  different machine architectures to interpret a given set in a consistent manner,
+     *  for serialize and deserialize operations.
+     */
+    for (uint16_t i = 0; i < mNextAvailable; i++)
+    {
+        Encoding::LittleEndian::Put64(reinterpret_cast<uint8_t *>(&mData[i]), mData[i]);
+    }
+
+    buflen = Base64Encode(reinterpret_cast<const uint8_t *>(mData), static_cast<uint16_t>(len), buf);
+    out    = buf;
+
+    for (uint16_t i = 0; i < mNextAvailable; i++)
+    {
+        mData[i] = Encoding::LittleEndian::HostSwap64(mData[i]);
```

Comment:
is there a way we can test that the Put64/HostSwap64 really are inverses of each other? I believe our unit tests only run on littleendian.

I would generally expect the same operation to be used for both save and revert, since if the architecture is a noop both are a noop and if the architecture swaps, both would swap.

Using Put64 in one case and HostSwap on the other make me more uneasy.